### PR TITLE
kinflate: Create AddConfigMap structure

### PR DESCRIPTION
**Pull request by apelisse**
This is empty and does nothing.

### CODE REVIEW

File: pkg/kinflate/commands/configmap.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+package commands
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+type AddConfigMap struct {
```
Comment (monopole) [inline pkg/kinflate/commands/configmap.go]: private?

File: pkg/kinflate/commands/configmap.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		RunE: func(_ *cobra.Command, args []string) error {
+			err := config.Validate()
+			if err != nil {
+				return err
+			}
+
+			if len(args) != 1 {
+				return fmt.Errorf("error: exactly one NAME is required, got %d", len(args))
+			}
+			config.Name = args[0]
```
Comment (monopole) [inline pkg/kinflate/commands/configmap.go]: suggest Validate accepts args in line 56 so you have something to validate.

also, please add test

File: pkg/kinflate/commands/configmap.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+type AddConfigMap struct {
+	// Name of configMap (required)
+	Name string
+	// FileSources to derive the configMap from (optional)
+	FileSources []string
+	// LiteralSources to derive the configMap from (optional)
+	LiteralSources []string
+	// EnvFileSource to derive the configMap from (optional)
+	EnvFileSource string
+}
```
Comment (monopole) [inline pkg/kinflate/commands/configmap.go]: suggest reference the Generic struct in pkg/apis/manifest/v1alpha1/types.go

File: pkg/kinflate/commands/configmap.go (hunk 2 above)
Comment (apelisse) [reply pkg/kinflate/commands/configmap.go]: Yeah, we'll add tests later

File: pkg/kinflate/commands/configmap.go (hunk 3 above)
Comment (apelisse) [reply pkg/kinflate/commands/configmap.go]: I don't think we want to strongly couple this structure with the definition of the manifest (nor have a comment that will be obsolete in the near future).

**Pull request merged**